src/main/ruby/truffleruby/core/string.rb

```diff
@@ -1097,6 +1097,82 @@ def rindex(sub, finish = undefined)
     nil
   end
 
+  def byteindex(str, start = undefined)
+    is_regex_pattern = Primitive.is_a?(str, Regexp)
+
+    if Primitive.undefined?(start)
+      start = 0
+    else
+      start = Primitive.rb_to_int(start)
+
+      start += bytesize if start < 0
+      if start < 0 || start > bytesize
+        if is_regex_pattern
+          Primitive.regexp_last_match_set(Primitive.caller_special_variables, nil)
+        end
+
+        return nil
+      end
+    end
+
+    unless Truffle::StringOperations.on_codepoint_boundary?(self, start)
+      raise IndexError, "offset #{start} does not land on character boundary"
+    end
+
+    if is_regex_pattern
+      Primitive.encoding_ensure_compatible(self, str)
+
+      match = Truffle::RegexpOperations.match_from(str, self, start)
+      Primitive.regexp_last_match_set(Primitive.caller_special_variables, match)
+      return match ? Primitive.character_index_to_byte_index(self, match.begin(0)) : nil
```

Comment:
This is inefficient, byte->char->byte offset. The whole point of these `byte` methods is to avoid conversion between byte/char offsets.
Could you use `Primitive.match_data_byte_begin(match, 0)` instead?

Reply:
Good to know, updated.